gwpy/timeseries/timeseries.py

```diff
@@ -1920,32 +1931,30 @@ def q_transform(self, qrange=(4, 64), frange=(0, numpy.inf),
 
         # build regular Spectrogram from peak-Q data by interpolating each
         # (Q, frequency) `TimeSeries` to have the same time resolution
-        nx = int(abs(Segment(*outseg)) / tres)
+        nx = int(abs(outseg) / tres)
         ny = frequencies.size
         out = Spectrogram(numpy.zeros((nx, ny)), x0=outseg[0], dx=tres,
                           frequencies=frequencies)
-        # FIXME: bug in Array2D.yindex setting
-        out._yindex = type(out.y0)(frequencies, out.y0.unit)
         # record Q in output
         out.q = peakq
         # interpolate rows
+        xout = numpy.linspace(0, 1, num=nx)
         for i, row in enumerate(norms):
-            interp = InterpolatedUnivariateSpline(
-                row.times.value, row.value)
-            out[:, i] = interp(out.times.value)
-
-        # then interpolate the spectrogram to increase the frequency resolution
-        # --- this is done because duncan doesn't like interpolated images
-        #     because they don't support log scaling
-        if fres is None:  # unless user tells us not to
+            xrow = numpy.linspace(0, 1, num=row.size)
+            interp = InterpolatedUnivariateSpline(xrow, row.value)
+            out[:, i] = interp(xout)
```

Comment:
@alurban, I don't understand what functional difference this change introduces? Can you explain this, please?

Reply:
@duncanmmacleod, this removes all references to a `Spectrogram.times` array and replaces them with a dummy variable, normalized between 0 and 1 so that the domain is consistent for interpolation. I find that this contributes to the reduction in memory overhead, as the production Omega scan code often reads in at least 64 seconds of data at a time per channel.

Reply:
I should clarify: doing this should have no (or very minor) impact on the output, but will reduce memory consumption for longer inputs.

Reply:
@duncanmmacleod, I had to go a different route to support using the `outseg` argument:

```python
# interpolate rows                                                      
xout = numpy.arange(outseg[0], outseg[1], tres)                         
for i, row in enumerate(norms):                                         
    xrow = numpy.arange(row.x0.value, (row.x0 + row.duration).value,    
                        row.dx.value)                                   
    interp = InterpolatedUnivariateSpline(xrow, row.value)              
    out[:, i] = interp(xout)
```

However, this still avoids an explicit call to `out.times` or `row.times`.

Reply:
Ok, thanks.